File: src/main/java/WordCounterApplication.java
import stopwords.StopWordsReader;
import stopwords.StopWordsReaderConfiguration;
import stopwords.StopWordsReaderImpl;

public class WordCounterApplication {

    private static final String STOP_WORDS_FILE_NAME = "stopwords.txt";

    public static void main(String[] args) {
        final StopWordsReader stopWordsReader = createStopWordsReader();

        final WordCounter wordCounter = new WordCounterImpl();

        final WordCounterService wordCounterService = new WordCounterServiceImpl(wordCounter, stopWordsReader);

        wordCounterService.countWords();
    }

    private static StopWordsReader createStopWordsReader() {
        final StopWordsReaderConfiguration stopWordsReaderConfiguration =
                new StopWordsReaderConfiguration(STOP_WORDS_FILE_NAME);

        return new StopWordsReaderImpl(stopWordsReaderConfiguration);
    }
}

File: src/main/kotlin/WordCounterApplication.kt
import java.io.File
import java.io.InputStreamReader

fun main(args: Array<String>) {
    val stopWords = readStopWordsFromClasspath()
    val counter = WordCounter(stopWords)

    val inputLine = if (args.isEmpty()) {
        readFromStdin()
    } else {
        readFromFile(args)
    }

    val (total, unique) = counter.countWords(inputLine)

    println("Number of words: $total, unique: $unique")
}

private fun readStopWordsFromClasspath(): Set<String> {
    val resourcePath = WordCounter::class.java.getResource("stopwords.txt") ?: error("Resource not found")
    val stopWords = resourcePath.openStream().use { inputStream ->
        InputStreamReader(inputStream).use { reader -> reader.readLines() }
    }.toSet()
    return stopWords
}

private fun readFromFile(args: Array<String>) = File(args.first()).readText()

private fun readFromStdin(): String {
    print("Enter text: ")
    return readln()
}
